File: backend/src/model/card/card.go
package card

import (
	"casino/model/card/rank"
	"casino/model/card/suit"
)

type Card struct {
	Suit suit.Suit
	Rank rank.Rank
}

func NewCard(suit suit.Suit, rank rank.Rank) *Card {
	return &Card{suit, rank}
}

File: frontend/src/utils/Card.ts
import {
  rankTable,
  RankKey,
  Rank,
  CardImage,
  Suit,
  Card,
  CardFromApi,
} from "../types/Card"

export const convertRank = (rank: RankKey): Rank => {
  return rankTable[rank]
}

export const convertCardImage = (card: Card): CardImage => {
  return `${card.suit.toLowerCase() as Lowercase<Suit>}_${
    card.rank.toLowerCase() as Lowercase<Rank>
  }`
}

export const toCard = (card: CardFromApi): Card => {
  return {
    rank: convertRank(card.Rank),
    suit: card.Suit.slice(0, -1) as Suit,
  }
}
